CONTRIBUTING.md

```diff
@@ -275,3 +275,59 @@ see <https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html>.
 
 With this, you should now have a working development setup! See
 [above](#building-and-testing-miri) for how to proceed working on Miri.
+
+### Syncing the miri subtree
+
+Note: Never rebase a subtree sync (in either direction). Always redo the operation.
+
+All of these operations happen *on a checkout of the rustc repo*. At no time do you
+even need a checkout of the miri repo or run any commands in it. 
+
+#### Moving changes from the rustc repo to the miri repo
+
+One time setup:
+
+Add your own miri fork as `miri-your-fork` to your rustc checkout:
+
+```
+git remote add miri-your-fork git@github.com:your-name/miri.git
+```
+
+Every time:
+
+Note: the first time you run this, `git subtree` builds a cache and this command may thusly take an hour or so.
```

Comment:
An hour?!??? :(

I have many rustc checkouts. How can I make it share the cache between my checkouts?

Reply:
I have no idea, this worked faster for clippy and other repos...

Reply:
If you use `git worktree` for creating the extra git checkouts I believe the subtree cache is shared. Not entirely sure though.

> I have no idea, this worked faster for clippy and other repos...

The first common commit between rust and miri is from 2018 (when https://github.com/rust-lang/rust/pull/46882 merged it into rustc as const eval engine). Git subtree has to walk all commits since then I believe.

Reply:
yea, that's not really useful... why doesn't it just walk the `subtree_add_commit..HEAD` range and rewrite those commits...